### ISSUE TITLE
Remove Host/Path From Component Uri In Database

### ISSUE THREAD

**Issue by jonwinton** [clay/clay#8]
## Description

Right now the data in the db is stored in the following format: `<host>/<path>/_<clay data structure>/...`

This means I can't restore data from one environment into another without a full data migration to swap out the `host/path` prefix. 

Stripping the host/path would get complicated in multi-site instances where components/pages follow a certain naming syntax. For example, say you have two sites `A` and `B` and both of them have a homepage. You may decide to make the following page uris: `a.com/_pages/homepage` and `b.com/_pages/homepage`. Without the host to namespace the instances you'd end up overwriting one or the other. 

## Proposal

For the problem listed above I think there are two possible solutions.  

1. Separate tables for the different sites
2. Prefix the keys with the site's slug instead of the host/path

My preference is to start with the second approach. This would make data portable between environments and you would only need to run a migration if you were changing the sites directory (`sites/<site_name>`).

## Questions

- What should the payload of PUT's be? I think all keys displayed at the JSON endpoints of Amphora should still show host/path but the Postgres module can swap the values in and out

## Impact
We should be able to scope this to just amphora-storage-postgres

**Comment by ncbooth** [comment]
An alternative approach would be not tying the site identifier and the path together.  If you store the path as a string on its own, and store the siteId as a separate field, and add a separate objectId field you wouldn't have to do translations.  To avoid collisions, you could add a multicolumn unique index on site_id & path.  Then in the amphora-storage-postgres module you append the "AND site_id = req.siteId" to every query.

**Comment by jonwinton** [comment]
Great call 👍 We'll add that to the approaches to test, but I think it makes a lot of sense.

@ncbooth do you think it's of value to require the JSON to still display the host and path when a dev does a request to a JSON endpoint of Amphora?

**Comment by ncbooth** [comment]
I think it makes sense for them to match the requested domain.  The clay-cli or amphora will have to strip them out and replace them with a siteId, right?

**Comment by jonwinton** [comment]
Yeah, it'll keep everything the exact same except for amphora-storage-postgres which will handle all the communication with the database while the other modules just interact with the REST API
